jobs/bump-jenkins-plugins.Jenkinsfile

```diff
@@ -0,0 +1,125 @@
+    def parts = pluginUrl.split("/")
+    def pluginVersion
+    if (parts.size() >= 4) {
+        def groupId = parts[-3]
+        pluginVersion = parts[-2]
+    } else {
+        error("Unable to extract plugin version from the URL.")
+    }
+    return pluginVersion
+}
+
+node {
+    checkout scm: [
+        $class: 'GitSCM',
+        branches: [[name: '*/pluginsupdate']],
+        userRemoteConfigs: [[url: "https://github.com/${fork_repo}.git"]],
+        extensions: [[$class: 'WipeWorkspace']]
+    ]
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+
+    properties([
+        pipelineTriggers([
+            // check once a month
+            pollSCM('H H 1 * *')
+        ]),
+        buildDiscarder(logRotator(
+            numToKeepStr: '100',
+            artifactNumToKeepStr: '100'
+        )),
+        durabilityHint('PERFORMANCE_OPTIMIZED')
+    ])
+
+    try {
+        shwrap("""
+            git config --global user.name "CoreOS Bot"
+            git config --global user.email "coreosbot-releng@fedoraproject.org"
+        """)
+
+        def pluginslist
+        def pluginsToUpdate = [:]
+        def plugins_lockfile = "jenkins/controller/plugins.txt"
+
+        stage("Read plugins.txt") {
+            shwrapCapture("""
+                git clone --branch pluginsupdate https://github.com/${fork_repo}.git
+
+                cd fedora-coreos-pipeline
+
+                # Check if the branch exists
+                if git ls-remote --heads --exit-code origin ${pr_branch} | grep ${pr_branch}; then
+                    git checkout ${pr_branch}
+                else
+                    git checkout -b ${pr_branch}
+                fi
+            """)
+            pluginslist = shwrapCapture("grep -v ^# ${plugins_lockfile}").split('\n')
+        }
+
+        stage("Check for plugin updates") {
+            def pluginUrl
+            pluginslist.each { plugin ->
+                def parts = plugin.split(':')
+                if (parts.size() == 2) {
+                    def pluginName = parts[0]
+                    def currentVersion = parts[1]
+                    pluginUrl = shwrapCapture("curl -Ls -I -f -o /dev/null -w '%{url_effective}' https://updates.jenkins.io/download/plugins/${pluginName}/latest/${pluginName}.hpi")
+                    def latestVersion = getVersionFromPluginUrl(pluginUrl)
+                    if (latestVersion.toString() != currentVersion.toString()) {
+                        pluginsToUpdate["${pluginName}"] = [currentVersion, latestVersion]
+                        println("Plugin: ${pluginName} current version is ${currentVersion}, it will be updated to latest version: ${latestVersion}")
+                        shwrap("""
+                            cd fedora-coreos-pipeline
+                            sed -i '/${pluginName}:/ s/${currentVersion}/${latestVersion}/g' ${plugins_lockfile}
```

Comment:
we should be able to do this find/replace once. maybe we just add this "pattern" to a list and then put them all together into a single sed command at the end?

i.e. the following works and processes two patterns:

```
sed -i -e 's/foo/bar' -e 's/abc/xyz' file.txt
```

Reply:
Changed the sed implementation here to add the plugins to a list and then make the sed operation on it. Kindly review the change.